core/src/main/java/org/jsmart/zerocode/core/httpclient/utils/UrlQueryParamsUtils.java

```diff
@@ -1,43 +1,26 @@
 package org.jsmart.zerocode.core.httpclient.utils;
 
-import org.apache.http.HttpEntity;
-import org.apache.http.NameValuePair;
-import org.apache.http.client.entity.UrlEncodedFormEntity;
-import org.apache.http.message.BasicNameValuePair;
-import org.apache.http.util.EntityUtils;
+import org.apache.http.client.utils.URIBuilder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.IOException;
-import java.util.ArrayList;
+import java.net.URISyntaxException;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
 import static java.util.Optional.ofNullable;
 
 public class UrlQueryParamsUtils {
     private static final Logger LOGGER = LoggerFactory.getLogger(UrlQueryParamsUtils.class);
 
-    public static String setQueryParams(String httpUrl, Map<String, Object> queryParams) throws IOException {
-        String qualifiedQueryParams = createQualifiedQueryParams(queryParams);
-        httpUrl = httpUrl + "?" + qualifiedQueryParams;
-
-        LOGGER.info("### Effective url is : " + httpUrl);
-        return httpUrl;
+    public static String setQueryParams(final String httpUrl, final Map<String, Object> queryParams) throws URISyntaxException {
+        URIBuilder uriBuilder = new URIBuilder(httpUrl);
+        Map<String, Object> nullSafeQueryParams = ofNullable(queryParams).orElseGet(HashMap::new);
+        nullSafeQueryParams.keySet().forEach(key ->
+                uriBuilder.addParameter(key, nullSafeQueryParams.get(key).toString())
+        );
+        String composedURL = uriBuilder.build().toString();
+        LOGGER.info("### Effective url is : {}", composedURL);
+        return composedURL;
     }
-
-    protected static String createQualifiedQueryParams(Map<String, Object> queryParamsMap) throws IOException {
```

Comment:
btw It's much more simple and we don't need this method

Reply:
Makes sense.  Looks much better now! Thank you 👍